File: API/gis/src/main/java/pt/isel/ps/gis/hypermedia/siren/components/subentities/Field.java
package pt.isel.ps.gis.hypermedia.siren.components.subentities;

import com.fasterxml.jackson.annotation.JsonInclude;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class Field {

    public enum Type {
        hidden, text, search, tel, url, email, password, datetime,
        date, month, week, time, datetimeLocal, number, range, color,
        checkbox, radio, file
    }

    private String name;
    private Type type;
    private Object value;

    public Field(String name, Type type, Object value) {
        this.name = name;
        this.type = type;
        this.value = value;
    }

    public String getName() {
        return name;
    }

    public Type getType() {
        return type;
    }

    public Object getValue() {
        return value;
    }
}

File: Android/gis/app/src/main/java/ps/leic/isel/pt/gis/hypermedia/subentities/Field.kt
package ps.leic.isel.pt.gis.hypermedia.subentities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@JsonIgnoreProperties(ignoreUnknown = true)
class Field(val name: String, val type: Type?, val value: Any?, var title: String?) {

    enum class Type private constructor(private val type: String) {
        hidden("hidden"), text("text"), search("search"), tel("tel"), url("url"), email("email"), password("password"),
        datetime("datetime"), date("date"), month("month"), week("week"), time("time"), datetimeLocal("datetimeLocal"),
        number("number"), range("range"), color("color"), checkbox("checkbox"), radio("radio"), file("file"),
        bool("boolean");

        override fun toString(): String {
            return type
        }
    }
}
